src/boardgameprojectfile/boardgameprojectfile.cpp

```diff
@@ -0,0 +1,38 @@
+#include "boardgameprojectfile.h"
+#include <QFile>
+#include <QDataStream>
+#include <QString>
+#include <QDebug>
+
+#include "bgame_generated.h"
+
+std::unique_ptr<BoardGame> BoardGameProjectFile::load(const QString &path)

```

Comment:
This method should return something meaningful or remove it completely.
Why it has a return type, but does not return anything? Do you think it is ok? It will cause a crash!